agents/agents-core/src/commonMain/kotlin/ai/koog/agents/core/dsl/builder/AIAgentSubgraphBuilder.kt

```diff
@@ -137,3 +159,35 @@ public open class AIAgentSubgraphDelegate<Input, Output> internal constructor(
         return subgraph!!
     }
 }
+
+/**
+ * Builder for a node that executes multiple nodes in parallel.
+ *
+ * @param nodes List of nodes to execute in parallel
+ * @param reduce Function to merge the contexts after parallel execution
+ */
+@OptIn(ExperimentalUuidApi::class)
+public class ParallelAIAgentNodeBuilder<Input, Output> internal constructor(
+    private val nodes: List<AIAgentNodeBase<Input, Output>>,
+    private val reduce: suspend AIAgentContextBase.(List<Triple<String, AIAgentContextBase, Output>>) -> Pair<AIAgentContextBase, Output>,
+) : AIAgentNodeBuilder<Input, Output>(
+    execute = { input ->
+        val initialContext: AIAgentContextBase = this
+        val (context, output) = supervisorScope {
+            val mapResults = nodes.map { node ->
+                async {
+                    val nodeContext = initialContext.copy()
+                    val result = node.execute(nodeContext, input)
+                    Triple(node.name, nodeContext, result)
+                }
+            }
+
+            val results = mapResults.awaitAll()
+
+            reduce(results)
```

Comment:
I think it would be good to provide ability to schedule such parallel execution on user provided Dispatcher, with default = Dispatchers.Default